templates/teachers_observations.html

```diff
@@ -99,6 +99,92 @@
     registerUpdateConstructsListener('#id_date_from');
     registerUpdateConstructsListener('#id_date_to');
 
+    function loadVideo(id, key) {
+        var $video = $('#' + key + '-source-' + id)[0];
+        if ($video) {
+            var link = $video.value;
+            var $videoContainer = $('#' + key + '-' + id)[0];
+            var $videoElement = $('<video />', {
+                style: 'width: 100%; height: 100%;',
+                src: link,
+                type: 'video/mp4',
+                controls: true,
+                preload: false,
+            })[0];
+
+            $videoContainer.append($videoElement);
+            $video.remove();
+        }
+    }
+
+    function loadPicture(id, is_touch_device) {
+        var $canvas = $('#the-modal .modal-body #' + id);
+        $canvas.attr('id', 'canvas' + id);
+
+        var annotation_data = $('#annotation_data_' + id).html();
+
+        $('#the-modal').on('shown.bs.modal', function() {
+            var url = $('#image-source-'+id).html();
+            if (!url) {
+                return;
+            }
+
+            var canvas = new fabric.Canvas('canvas' + id, {
+                isDrawingMode: false,
+                selection: false,
+            });
+
+            const canvasHeight = 332;
+            const canvasWidth = 530;
+
+            var bgImg = new fabric.Image();
+
+            bgImg.setSrc(url.replace(/&amp;/g, '&'), function(obj) {
+                var imageWidth = $('#image-width_' + id).html();
+
+                if (imageWidth == 'None') {
+                    imageWidth = 500
```

Comment:
Should it be 500 here?

Reply:
I change it to 530 because canvas width is 530. And image will fit properly.

Reply:
Am what?